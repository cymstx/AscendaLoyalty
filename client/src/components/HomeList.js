import React from "react";
import { Button } from "antd";
import { useNavigate } from "react-router-dom";
import "./styles/clientRewards.scss";

const HomeList = (props) => {
  const navigate = useNavigate();
  var button_id = "bank-" + props.id;
  return (
    <>
      <Button
        id={button_id}
        onClick={() => navigate("/bank-" + props.id)}
        type="primary"
        className="card_rewards__btn hover"
      >
        View Available Rewards
      </Button>
    </>
  );
};

export default HomeList;
