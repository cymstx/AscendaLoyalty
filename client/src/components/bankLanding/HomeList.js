import React from "react";
import { Button } from "antd";
import { Carousel } from "antd";
import { Col, Row } from "antd";
import { useNavigate } from "react-router-dom";
import "./bank.css";
import nationalday from "../images/nationalday.jpg";
import test from "../images/test.jpg";

const contentStyle = {
  height: "200px",
  color: "#fff",
  lineHeight: "50px",
  textAlign: "center",
  background: "#b82828",
};

const user = localStorage.getItem("user");

const HomeList = (props) => {
  const navigate = useNavigate();
  var button_id = "bank-" + props.id;
  return (
    <div className="backgnd">
      <div>
        <h2 className="greeting">Welcome Back,</h2>
        <Row>
          <Col span={12}>
            <h1 className="user">{user}</h1>
            <div className="carousel">
              <Carousel autoplay speed={3500}>
                <div>
                  <h3 style={contentStyle}>
                    Emails and SMSes from DBS will not include clickable links.
                    Always go directly to our website to verify the authenticity
                    of any banking-related requests or offers.
                  </h3>
                </div>
                <div>
                  <h3 style={contentStyle}>
                    DBS will never ask you for your credit or debit card
                    details, CVV, SMS or email OTPs, or Digital Token approvals
                    to verify or unlock your account.
                  </h3>
                </div>
                <div>
                  <h3 style={contentStyle}>
                    Do not call phone numbers, click on URL links, or scan QR
                    codes in unsolicited emails, SMS, or other Messaging
                    Application messages.
                  </h3>
                </div>
                <div>
                  <h3 style={contentStyle}>
                    Never disclose your card numbers or OTPs to unverified
                    sources. Bank staff and government officials will never
                    request your card details, OTPs, or Digital Token Approvals
                    through SMS, voice calls, or unofficial websites.
                  </h3>
                </div>
              </Carousel>
              <h3>
                Here at DBS, we wish Singapore a very happy{" "}
                <b>
                  57<sup>th</sup>
                </b>{" "}
                birthday!
              </h3>
              <h3 className="h3">
                Enjoy this festive National Day by claiming some of your points
                at a higher rate!
              </h3>
              <h3>
                Special offers start on <b>1/8/2022</b> and ends on{" "}
                <b>6/9/2022</b>. Get yours now!
              </h3>
              <Button
                id={button_id}
                onClick={() => navigate("/bank-" + props.id)}
                type="danger"
                className="card_rewards__btn hover"
              >
                View Available Rewards
              </Button>
            </div>
          </Col>
          <Col span={12}>
            {/* <img src={nationalday} className="image" /> */}
            <img src={test} className="image" />
          </Col>
        </Row>
      </div>
    </div>
  );
};

export default HomeList;
