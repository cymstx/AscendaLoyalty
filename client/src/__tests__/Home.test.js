import React from 'react';
import '../matchMedia.js';
import HomeList from '../components/bankLanding/HomeList';
import renderer from 'react-test-renderer'
import {BrowserRouter as Router} from 'react-router-dom';

describe("snapshot testing", () => {
    test("snapshot for Home component", () => {
        const renderedComponent = renderer.create(
        <Router>
            <HomeList />
        </Router>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
}) 