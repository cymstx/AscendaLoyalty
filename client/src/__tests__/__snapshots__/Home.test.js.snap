// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Home component 1`] = `
<div
  className="backgnd"
>
  <div>
    <h2
      className="greeting"
    >
      Welcome Back,
    </h2>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col ant-col-12"
        style={Object {}}
      >
        <h1
          className="user"
        />
        <div
          className="carousel"
        >
          <div
            className="ant-carousel"
          >
            <div
              className="slick-slider slick-initialized"
              dir="ltr"
            >
              <div
                className="slick-list"
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={null}
                onMouseMove={null}
                onMouseUp={[Function]}
                onTouchCancel={null}
                onTouchEnd={[Function]}
                onTouchMove={null}
                onTouchStart={[Function]}
                style={Object {}}
              >
                <div
                  className="slick-track"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "WebkitTransform": "translate3d(0px, 0px, 0px)",
                      "WebkitTransition": "",
                      "msTransform": "translateX(0px)",
                      "opacity": 1,
                      "transform": "translate3d(0px, 0px, 0px)",
                      "transition": "",
                    }
                  }
                >
                  <div
                    aria-hidden={true}
                    className="slick-slide slick-cloned"
                    data-index={-1}
                    onClick={[Function]}
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Never disclose your card numbers or OTPs to unverified sources. Bank staff and government officials will never request your card details, OTPs, or Digital Token Approvals through SMS, voice calls, or unofficial websites.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={false}
                    className="slick-slide slick-active slick-current"
                    data-index={0}
                    onClick={[Function]}
                    style={
                      Object {
                        "outline": "none",
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Emails and SMSes from DBS will not include clickable links. Always go directly to our website to verify the authenticity of any banking-related requests or offers.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide"
                    data-index={1}
                    onClick={[Function]}
                    style={
                      Object {
                        "outline": "none",
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          DBS will never ask you for your credit or debit card details, CVV, SMS or email OTPs, or Digital Token approvals to verify or unlock your account.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide"
                    data-index={2}
                    onClick={[Function]}
                    style={
                      Object {
                        "outline": "none",
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Do not call phone numbers, click on URL links, or scan QR codes in unsolicited emails, SMS, or other Messaging Application messages.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide"
                    data-index={3}
                    onClick={[Function]}
                    style={
                      Object {
                        "outline": "none",
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Never disclose your card numbers or OTPs to unverified sources. Bank staff and government officials will never request your card details, OTPs, or Digital Token Approvals through SMS, voice calls, or unofficial websites.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide slick-cloned"
                    data-index={4}
                    onClick={[Function]}
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Emails and SMSes from DBS will not include clickable links. Always go directly to our website to verify the authenticity of any banking-related requests or offers.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide slick-cloned"
                    data-index={5}
                    onClick={[Function]}
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          DBS will never ask you for your credit or debit card details, CVV, SMS or email OTPs, or Digital Token approvals to verify or unlock your account.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide slick-cloned"
                    data-index={6}
                    onClick={[Function]}
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Do not call phone numbers, click on URL links, or scan QR codes in unsolicited emails, SMS, or other Messaging Application messages.
                        </h3>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-hidden={true}
                    className="slick-slide slick-cloned"
                    data-index={7}
                    onClick={[Function]}
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                    tabIndex="-1"
                  >
                    <div>
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                      >
                        <h3
                          style={
                            Object {
                              "background": "#b82828",
                              "color": "#fff",
                              "height": "200px",
                              "lineHeight": "50px",
                              "textAlign": "center",
                            }
                          }
                        >
                          Never disclose your card numbers or OTPs to unverified sources. Bank staff and government officials will never request your card details, OTPs, or Digital Token Approvals through SMS, voice calls, or unofficial websites.
                        </h3>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <ul
                className="slick-dots slick-dots-bottom"
                onMouseEnter={null}
                onMouseLeave={null}
                onMouseOver={null}
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <li
                  className="slick-active"
                >
                  <button
                    onClick={[Function]}
                  >
                    1
                  </button>
                </li>
                <li
                  className=""
                >
                  <button
                    onClick={[Function]}
                  >
                    2
                  </button>
                </li>
                <li
                  className=""
                >
                  <button
                    onClick={[Function]}
                  >
                    3
                  </button>
                </li>
                <li
                  className=""
                >
                  <button
                    onClick={[Function]}
                  >
                    4
                  </button>
                </li>
              </ul>
            </div>
          </div>
          <h3>
            Here at DBS, we wish Singapore a very happy
             
            <b>
              57
              <sup>
                th
              </sup>
            </b>
             
            birthday!
          </h3>
          <h3
            className="h3"
          >
            Enjoy this festive National Day by claiming some of your points at a higher rate!
          </h3>
          <h3>
            Special offers start on 
            <b>
              1/8/2022
            </b>
             and ends on
             
            <b>
              6/9/2022
            </b>
            . Get yours now!
          </h3>
          <button
            className="ant-btn ant-btn-danger card_rewards__btn hover"
            disabled={false}
            id="bank-undefined"
            onClick={[Function]}
            type="button"
          >
            <span>
              View Available Rewards
            </span>
          </button>
        </div>
      </div>
      <div
        className="ant-col ant-col-12"
        style={Object {}}
      >
        <img
          className="image"
          src="nationalday.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
